packages/zarrita/src/codecs/fixedscaleoffset.ts

```diff
@@ -0,0 +1,56 @@
+import type { Chunk, NumberDataType, TypedArrayConstructor } from "../metadata.js"
+import { coerce_dtype, get_ctr } from "../util.js"
+
+type FixedScaleOffsetConfig = {
+    offset: number;
+    scale: number;
+    dtype: string;
+    astype?: string;
+}
+
+export class FixedScaleOffsetCodec<D extends NumberDataType, A extends NumberDataType> {
+    readonly kind = "array_to_array";
+
+    #offset: number;
+    #scale: number;
+    #TypedArrayIn: TypedArrayConstructor<D>
+    #TypedArrayOut: TypedArrayConstructor<A>
+
+    constructor(configuration: FixedScaleOffsetConfig) {
+        const { data_type } = coerce_dtype(configuration.dtype);
```

Comment:
testing on v3 has made me realize that `coerce_dtype` is really only meant for v2 data type strings, so this will need to be adjusted.